src/modules/emote_menu/components/Menu.jsx

```diff
@@ -0,0 +1,143 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import Divider from 'rsuite/lib/Divider/index.js';
+import keycodes from '../../../utils/keycodes.js';
+import emoteStore from '../stores/index.js';
+import styles from '../styles/menu.module.css';
+import Emotes from './Emotes.jsx';
+import Header from './Header.jsx';
+import Preview from './Preview.jsx';
+import Sidebar from './Sidebar.jsx';
+
+let altKeyPressed = false;
+let shiftKeyPressed = false;
+
+export default function EmoteMenu({triggerRef, appendToChat}) {
+  const [search, setSearch] = useState('');
+  const [selected, setSelected] = useState(null);
+  const [, setUpdated] = useState(false);
+
+  const [section, setSection] = useState({
+    eventKey: null,
+    scrollTo: false,
+  });
+
+  const [arrowKeys, setArrowKeys] = useState({
+    top: false,
+    down: false,
+    right: false,
+    left: false,
+  });
+
+  const onHide = useCallback(() => triggerRef.current.close(), [triggerRef]);
+
+  const handleClick = useCallback((emote) => {
+    if (altKeyPressed) {
+      emoteStore.toggleFavorite(emote);
+      return;
+    }
+
+    appendToChat(emote.code);
+    emoteStore.trackHistory(emote);
+
+    if (shiftKeyPressed) {
+      return;
+    }
+
+    onHide();
+  }, []);
+
+  useEffect(() => setSearch(''), [section]);
+
+  useEffect(() => {
+    function buttonPressCallback(event) {
+      altKeyPressed = event.altKey;
```

Comment:
should we be storing these states with a `useState` hook? multiple mounts of a component could cause these variables to be unexpectedly changed during runtime.